pkg/controller/podautoscaler/metrics/metrics_client.go

```diff
@@ -137,23 +158,30 @@ func (h *HeapsterMetricsClient) GetCpuConsumptionAndRequestInMillis(namespace st
 			}
 		}
 	}
-	if len(podNames) == 0 && len(podList.Items) > 0 {
-		return 0, 0, 0, time.Time{}, fmt.Errorf("no running pods")
+	if readyPods.Len() == 0 {
+		return 0, 0, nil, fmt.Errorf("no running and ready pods")
 	}
 	if missing || requestSum == 0 {
-		return 0, 0, 0, time.Time{}, fmt.Errorf("some pods do not have request for cpu")
+		return 0, 0, nil, fmt.Errorf("some pods do not have request for cpu")
```

Comment:
hmm... I'd agree for missing *metrics* but for missing *request*, we should probably just error out (because that's easy for the user to fix.  I'll make missing metrics do this.